### ISSUE TITLE
Amazon returns only 100 orders per call

### ISSUE THREAD

**Issue by Johnny99211** [jlevers/selling-partner-api#308]
## Problem description:

I'm wondering why Amazon only returns 100 orders within one call? I'm requesting the orders for the last 5 days every midnight. The limit breaks my whole code. Any advice how to deal with this?

## Code

```php
$created_after = date( 'Y-m-d\TH:i:s', strtotime( '-5 days' ) );

$result = $this->orders_api->getOrders( [ $this->marketplace_id ], $created_after, null, null, null, null, null, null, null, null, null, null, null, [
	// Add order IDs here in case we want to test
], null, null, null, [
	'buyerInfo',
	'shippingAddress'
] );
```

**Comment by yourguide** [comment]
Yeah thats the max they return on each request of orders. You will find a TOKEN comes back with each request until you have them all. 
So the process is request your orders, get back the first 100, if you get a Token in the response, send that token to the next request and repeat until you have all the orders.
Hope that helps.

**Comment by Johnny99211** [comment]
@yourguide thanks a lot for your response. Do you have any code you can share with me? Because I'm wondering where I should add the token to each request? Since I'm working on a customer's Prod system, I'm worried trying. If you already did this, it would be a huge help!

**Comment by yourguide** [comment]
In the documentation for the library you should see this line:
//$next_token = ''; // string | A string token returned in the response of your previous request.

It's default value is '' or you can make it NULL also.... you send it with the request like this: 
```
$result = $apiInstance->getOrders($marketplace_ids, $created_after, $created_before, $last_updated_after, $last_updated_before, $order_statuses, $fulfillment_channels, $payment_methods, $buyer_email, $seller_order_id, $max_results_per_page, $easy_ship_shipment_statuses, $next_token, $amazon_order_ids, $actual_fulfillment_supply_source_id, $is_ispu, $store_chain_store_id, $data_elements);
```
So when you get the response back I just do this with it:
```
	$OrdersJSON = json_encode($result->jsonSerialize());
	$OrdersArray = json_decode($OrdersJSON, true);
```
and then I check that array for the existence of the  "NextToken" and put it in a while loop that keeps calling the get orders until there isn't one like this:
```
			while(isset($OrdersArray['payload']['NextToken'])){
                                 // call get orders function
                                     }
```

**Comment by Johnny99211** [comment]
@yourguide awesome! Thanks for the hint. I'll try it out and let you know if it works. I'll post the whole solution as well.

**Comment by Johnny99211** [comment]
Got it to work. I've wrote myself a little function which is responsible for the request:

`/**
 * Request the orders
 *
 * @param string $created_after
 * @param string|null $next_token
 *
 * @return false|GetOrdersResponse
 */
private function request_orders( string $created_after, string $next_token = null ) {
	try {
		return $this->orders_api->getOrders( [ $this->marketplace_id ], $created_after, null, null, null, null, null, null, null, null, null, null, $next_token, [
			// Add order IDs here in case we want to test
		], null, null, null, [
			'buyerInfo',
			'shippingAddress'
		] );
	} catch ( ApiException $e ) {
		return false;
	}
}`

I'm calling the function this way:

`/**
 * Returns all available orders from Amazon
 *
 * @param string $created_after
 *
 * @return Order[]|false
 */
public function get_orders( string $created_after ) {
	try {
		$result = $this->request_orders( $created_after );

		if ( empty( $result->getErrors() ) && $result->getPayload() !== null ) {
			$orders = [];

			$orders = array_merge( $result->getPayload()->getOrders(), $orders );

			// We have another orders to request - continue
			while ( ! empty( $result->getPayload()->getNextToken() ) ) {
				$result = $this->request_orders( $created_after, $result->getPayload()->getNextToken() );

				if ( empty( $result->getErrors() ) && $result->getPayload() !== null ) {
					$orders = array_merge( $result->getPayload()->getOrders(), $orders );
				}

				// Sleep for 3 second because otherwise we will get limit warning from Amazon
				sleep( 3 );
			}

			return $orders;
		}

		return false;
	} catch ( Exception $e ) {
		return false;
	}
}`

The above function is my main function which will be called initially.
